platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfigurationContainer.java

```diff
@@ -114,15 +115,15 @@ protected NamedDomainObjectProvider<Configuration> createDomainObjectProvider(St
     @Override
     public void visitConsumable(Consumer<ConfigurationInternal> visitor) {
         // Visit all configurations which are known to be consumable
-        withType(ConsumableConfiguration.class).forEach(configuration ->
+        Collection<ConsumableConfiguration> availableConsumableConfigurations = new ArrayList<>(withType(ConsumableConfiguration.class));
```

Comment:
Can we add a comment here explaining why we copy before visiting?

I'd like there to be enough info that someone looking at this code understands why we are doing this extra step.

My immediate reaction seeing this code without context would be to "fix" it